subprojects/model-core/src/main/java/org/gradle/api/internal/provider/CredentialsProviderFactory.java

```diff
@@ -155,6 +161,25 @@ public synchronized AwsCredentials call() {
         }
     }
 
+    private class HttpHeaderCredentialsProvider extends CredentialsProvider<HttpHeaderCredentials> {
+
+        HttpHeaderCredentialsProvider(String identity) {
+            super(identity);
+        }
+
+        @Override
+        public synchronized HttpHeaderCredentials call() {
+            String name = getRequiredProperty("Name");
+            String value = getRequiredProperty("Value");
```

Comment:
I agree, but this would make it inconsistent with the name of the properties in `HttpHeaderCredentials`, wouldn't it? The [current docs](https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:handling_credentials) seem to suggest that though it is not mentioned explicitly, but I guess the docs could be changed to signal the different name.

Reply:
Done, but I'm not a fan of the way it's documented, but I don't know where to put that information otherwise 🤔

Reply:
Thanks! I'll rewrite the user guide to reflect that as well.